composer/loggers/mosaicml_logger.py

```diff
@@ -88,13 +90,30 @@ def log_metrics(self, metrics: Dict[str, Any], step: Optional[int] = None) -> No
         self._log_metadata(metrics)
 
     def after_load(self, state: State, logger: Logger) -> None:
+        # Log model data downloaded and initialized for run events
+        self._log_metadata({'model_initialized_time': time.time()})
         # Log WandB run URL if it exists. Must run on after_load as WandB is setup on event init
         for callback in state.callbacks:
             if isinstance(callback, WandBLogger):
                 run_url = callback.run_url
                 if run_url is not None:
                     self._log_metadata({'wandb/run_url': run_url})
 
+    def fit_start(self, state: State, logger: Logger) -> None:
+        # Log max duration in either epochs or tokens for run events
+        if state.max_duration.unit == TimeUnit.TOKEN:
+            self._log_metadata({'total_num_tokens': state.max_duration.value})
+        else:
+            num_batches_per_epoch = math.ceil(len(state.train_dataloader.dataset) / state.train_dataloader.batch_size)
```

Comment:
This requires more try/catching if left this way. datasets are not guaranteed to have length, and dataloaders are not guaranteed to have batch size. I suggest doing the above suggestion, does that still work for your event needs?

Reply:
Hmm I see what you mean, do you have an idea how often the dataset length and batch size are missing? It would be great to report [epoch x/xx] [batch x/xx] instead of just [epoch x/xx] but if this can't be reported often, then I think we should standardize to max duration and unit.